### PR TITLE
Drop old pending* aliases from QgsVectorLayer

### PR DESCRIPTION
**Pull request by nyalldawson**
Drops the old pendingFields(), pendingFeatureCount(), pendingAllAttributesList() and pendingPkAttributesList() methods from the QgsVectorLayer API. The fields(), featureCount(), allAttributes() and pkAttributeList() methods should be used instead.

### CODE REVIEW

File: doc/api_break.dox [hunk 1]
```diff
@@ -2431,6 +2431,10 @@ displayExpression instead. For the map tip use mapTipTemplate() instead.
 - addPart() now returns QgsGeometry::OperationResult enum, integer representation of returned values may have changed
 - splitParts() now returns QgsGeometry::OperationResult enum, integer representation of returned values may have changed
 - splitFeatured() now returns QgsGeometry::OperationResult enum, integer representation of returned values may have changed
+- pendingFields() was dropped. Use fields() instead.
+- pendingAllAttributesList() was dropped. Use allAttributes() instead.
+- pendingPkAttributesList() was dropped. Use pkAttributeList() instead.
```
Comment (m-kuhn) [inline doc/api_break.dox]: Can this be renamed to pkAttributes() for consistency with allAttributes?

**Review by m-kuhn — APPROVED** [review]
Love it

File: doc/api_break.dox (hunk 1 above)
Comment (nyalldawson) [reply doc/api_break.dox]: Done - I've renamed it to primaryKeyAttributes() to get rid of the acronym too.

**Comment by NathanW2** [comment]
NIce!

**Pull request merged**